### DISCUSSION TITLE
Improvement of `resolvePageDate`

### DISCUSSION THREAD

**Discussion by Mister-Hope** [vuepress/core#46] [Ideas]
Currently, resolvePageDate is returning `1970-01-01` when the Date is empty or invaid. And the date is use to built permalinks.

https://github.com/vuepress/vuepress-next/blob/55f780c4a516b2c0119328874906647af80ac73b/packages/%40vuepress/core/__tests__/page/resolvePageDate.spec.ts#L27-L66

And in that case, if set premalink with date globally, we will get links like below with `/1970/01/01/foo-bar/frontmatter` .

https://github.com/vuepress/vuepress-next/blob/55f780c4a516b2c0119328874906647af80ac73b/packages/%40vuepress/core/__tests__/page/resolvePagePermalink.spec.ts#L48-L61

I think that's not good enough, visitors would be confused if they are not familiar with coding. 

My ideas is to let them fill with `__`(or `x` `-` `_` or something else).  so that if global permalinks with date are set, those without valid date would be set as `/__/__/__/foo-bar/`. I think this can be much more clear to site owners or visitors to release that it's not having date. 😀

---

Also I would suggest date without day like below return with `2020-10-__`

https://github.com/vuepress/vuepress-next/blob/55f780c4a516b2c0119328874906647af80ac73b/packages/%40vuepress/core/__tests__/page/resolvePageDate.spec.ts#L77-L86

So that we can built permalinks like `/2020/10/__/foo-bar/`, looking forward to your opinion.

**Comment by meteorlxy** [comment]
That looks so weird.

And will complicate the logic

**Reply by Mister-Hope** [reply]
I think for thoses who do not code, a link like `/_/_/_/foo-bar/` would be better then ` /1970/01/01/foo-bar/`. I asked some of my friends. 5 of 6 agree that the fist link is more meanful. The most question is for my friends who majored in physics, 1970 01 01 is a date they can not understand. And all of them are requesting to stay with `/foo-bar/`, (which may confict with the original link, out of my thought)

I am just suggesting if we can change 1970-01-01 to something else. I am hoping we can let people understand that this link is not containing a date, while remain not conficting the original links.

**Reply by Mister-Hope** [reply]
This parts of code changing is just affecting permalinks now, and at least in V1 it's not used in other places. So I think some special handling is ok.😑

**Reply by meteorlxy** [reply]
What about 0000-00-00 ?

**Reply by Mister-Hope** [reply]
I think it's better, let me ask my friend about it

**Reply by Mister-Hope** [reply]
3 of them replied and agreed, this won't let them confuse if it's a acticle that do write in 1970.😀

**Reply by meteorlxy** [reply]
Change to `0000-00-00` in v2.0.0-alpha.24
